### ISSUE TITLE
Premature global checkpoint advancement & local checkpoint regression after dispatching table

### ISSUE THREAD

**Issue by liuzix** [pingcap/ticdc#1330]
Consider the following scenario:
1. Processor has local checkpoint ts `t1`, and the global checkpoint ts is `t0`, where `t1 > t0`.
2. Owner dispatches a table whose start-ts is `t0` to said processor.
3. Processor starts tableSink for the newly added table, which reports a table level checkpoint `t0`, i.e. the start-ts. This will lead to a local checkpoint regression to `t0`.
4. Processor marks the operation as done before the table level checkpoint catches up with `t1`.
5. Owner advances global checkpoint to `t1`, because processor has yet to upload the regressed local checkpoint.
6. Processor updates the local checkpoint `t0`.

To sum up, prior to the series of events, we have global checkpoint `t0` and local checkpoint `t1` where `t1 > t0`. After the events, however, the two checkpoints change places with each other, a very unexpected outcome.

This bug does not seem to affect any released version.
